packages/gatsby/src/schema/infer/is-file.js

```diff
@@ -0,0 +1,85 @@
+const _ = require(`lodash`)
+const path = require(`path`)
+const slash = require(`slash`)
+const mime = require(`mime`)
+const isRelative = require(`is-relative`)
+const isRelativeUrl = require(`is-relative-url`)
+
+const isFile = (nodeStore, field, relativePath) => {
+  const filePath = getFilePath(nodeStore, field, relativePath)
+  if (!filePath) return false
+  const filePathExists = nodeStore
+    .getNodesByType(`File`)
+    .some(node => node.absolutePath === filePath)
+  return filePathExists
+}
+
+module.exports = {
+  isFile,
+}
+
+const getFirstValueAt = (node, selector) => {
+  let value = _.get(node, selector)
```

Comment:
I did not use `_.get` but `getValueAtSelector` because lodash does not work with arrays of objects - we can make certain assumptions in a graphql context that a more general solution like lodash cannot: that arrays are lists with uniform types.